src/common/harvest/harvest.js

```diff
@@ -88,7 +88,7 @@ export class Harvest extends SharedContext {
 
     var agentRuntime = getRuntime(this.sharedContext.agentIdentifier)
 
-    if (!payload.body) { // no payload body? nothing to send, just run onfinish stuff and return
+    if (!payload.body && !payload.qs) { // no payload body? nothing to send, just run onfinish stuff and return
```

Comment:
Do we have cases where a request can be sent without a body?

Reply:
Why's payload w/ `qs` allowed through?

Reply:
Before this change, the page view event aggregator called directly to the send data module. I would like to remove that direct tying and just use the harvest module but if there are no custom attributes, then the body is empty for the rum post call. How to handle that?

Reply:
Theres some ways but IDK if I'm in love with any of them.  Could set up rules for each endpoint for instance, which might be needed for other features down the line re: retries, codes body vs query, etc

Reply:
Though I get making PVE "in line" with the others, not sure it's worth it to create a "harvest scheduler" for something only sent once on load then becomes irrelevant, unless the query params are moving to the body too.
(Metrics now only sent once too, but it re-uses the harvest's EoL logic & does send stuff in body.)

Reply:
Maybe rum consumer accepts & does nothing with an empty `{}` object for body

Reply:
The logic in the harvest module is to not send a body if it is empty. That would also affect every other endpoint.